### ISSUE TITLE
Crash on A1200 BlizzardPPC + Kickstart 3.1.4

### ISSUE THREAD

**Issue by bSr43** [keirf/Amiga-Stuff#16]
First, thank you very much for such a great utility! I use it very often, and it works very well. It has helped me in finding many hardware issues on many of my Amiga!

Nevertheless, AmigaTestKit crashes on my Amiga 1200 equipped with BlizzardPPC/060 (FPU+MMU) and a Kickstart 3.1.4.
The crash can be reproduced on a similar setup on FS-UAE (when using a full 68060 CPU with MMU, not the default one which comes without MMU).

After digging into the issue, it appears that it is not possible in this hardware configuration to copy anything in the memory range 0x400-0x3FFF. The copy loop in base/init.S ends up with only zeroes in this memory range, which makes the program crash at a later point.

I've read on EAB that Kickstart 3.1.4 reserves 16k of chip memory for emulators (to avoid tools like PrepareEmul for ShapeShifter) which may partly explain why ATK doesn't work, but the problem only exhibits when one also uses a BlizzardPPC with complete 68060.

I've compiled a version of ATK where I've changed the base address from 0x400 to 0x4000 in the file base/amiga.ld and now it works perfectly fine on FS-UAE and on my real computer.

**Comment by keirf** [comment]
I wonder if the best bet is to turn the mmu off. Forcibly. It might be unfortunate if we're running at a remapped virtual address but I don't think ATK would like that anyway!

**Comment by bSr43** [comment]
You're right. I have submitted a PR which disable MMU, and now it works as expected.

**Comment by keirf** [comment]
Can i assume it is a standard code block for mmu disable which, assuming exceptions are handled, is safe on any m68k cpu?

**Comment by bSr43** [comment]
It was only disabling 040 and 060 MMU. I've updated the PR so that it disables 68851 and 68030 MMU. Also, I've cleaned the 040/060 code because it was setting a useless transparent translation. Now it only disables everything.
I've tested this version on many of my real Amiga, covering 68000, 680EC020, 68030, 68040 and 68060. Unfortunately, I don't own any 68851, 68010, nor EC/LC versions of 040/060. So I tried on  FS-UAE and WinUAE, and it worked fine.
I'm pretty sure that this version should work with any CPU :)
